python_modules/airline-demo/airline_demo/solids.py

```diff
@@ -208,19 +215,22 @@ def download_from_s3(info):
     '''
     bucket = info.config['bucket']
     key = info.config['key']
-    target_path = (info.config.get('target_path') or key)
+    target_path = info.config.get('target_path')
 
-    if info.config['skip_if_present'] and os.path.isfile(target_path):
+    if target_path and info.config['skip_if_present'] and os.path.isfile(target_path):
         info.context.info(
             'Skipping download, file already present at {target_path}'.format(
                 target_path=target_path
             )
         )
         return target_path
 
-    if os.path.dirname(target_path):
+    if target_path and os.path.dirname(target_path):
         mkdir_p(os.path.dirname(target_path))
 
+    if target_path is None:
+        target_path = info.context.resources.tempfile.tempfile().name
```

Comment:
hmmm we might want to consider a more "functional" api here where the tempfile manager has a track less state

Reply:
yep, the thing i was trying to solve for here is the case where we want a named temporary file that persists across node execution boundaries. i'm very open to a more functional api that lets us keep the underlying NamedTemporaryFile around but i don't see the obvious change